java/ql/src/experimental/Security/CWE/CWE-117/LogInjection.qll

```diff
@@ -0,0 +1,84 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about untrusted user input used in log entries.
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+
+module LogInjection {
+  string logLevel() {
+    result = "trace" or
+    result = "info" or
+    result = "warn" or
+    result = "error" or
+    result = "fatal"
+  }
+
+  /**
+   * A data flow source for user input used in log entries.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for user input used in log entries.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for malicious user input used in log entries.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * A taint-tracking configuration for untrusted user input used in log entries.
+   */
+  class LogInjectionConfiguration extends TaintTracking::Configuration {
+    LogInjectionConfiguration() { this = "LogInjection" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    override predicate isSanitizer(DataFlow::Node node) { node instanceof Sanitizer }
+  }
+
+  /**
+   * A source of remote user controlled input.
+   */
+  class RemoteSource extends Source {
+    RemoteSource() { this instanceof RemoteFlowSource }
+  }
+
+  /**
+   * A class of popular logging utilities.
+   */
+  class LoggingType extends RefType {
+    LoggingType() {
```

Comment:
Would probably be good to cover the following ones as well:
- Log4j1: [`org.apache.log4j.Category`](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/Category.html)
- Log4j2: [`org.apache.logging.log4j.LogBuilder`](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/LogBuilder.html)
- SLF4J: [`org.slf4j.spi.LoggingEventBuilder`](http://www.slf4j.org/apidocs/org/slf4j/spi/LoggingEventBuilder.html)
- Apache Commons Logging: [`org.apache.commons.logging.Log`](https://commons.apache.org/proper/commons-logging/javadocs/api-release/org/apache/commons/logging/Log.html)
- JBoss Logging: [`org.jboss.logging.BasicLogger`](https://docs.jboss.org/jbosslogging/latest/org/jboss/logging/BasicLogger.html)

And with that many types it might be good to add comments here explaining which type belongs to which framework, since especially for Log4j1, Log4j2 and Apache Commons Logging it can become quite confusing.